tests/www/test_security.py

```diff
@@ -308,3 +308,26 @@ def test_override_role_vm(self):
         test_security_manager = TestSecurityManager(appbuilder=self.appbuilder)
         self.assertEqual(len(test_security_manager.VIEWER_VMS), 1)
         self.assertEqual(test_security_manager.VIEWER_VMS, {'Airflow'})
+
+    def test_is_user_logged_in_returns_false_if_not_authenticated(self):
+        user = mock.MagicMock()
+        user.is_authenticated = False
+        self.assertFalse(self.security_manager.is_user_logged_in(user))
+
+    def test_is_user_logged_in_returns_true_if_authenticated(self):
+        user = mock.MagicMock()
+        user.is_authenticated = True
+        self.assertTrue(self.security_manager.is_user_logged_in(user))
+
+    def test_unauthenticated_user_is_public(self):
+        user = mock.MagicMock()
+        user.is_authenticated = False
+        self.assertTrue(self.security_manager.is_public_user(user))
```

Comment:
I'm not sure I like this behaviour.

I think a better fix would be to change the Sign-up role. Anonymous users shouldn't pass any `@has_access` check to my mind.

Or this should be a config setting somewhere.

Reply:
Are you suggesting creating some role to which users are added by default on sign-up which has permissions only to the (empty) Index page? If so, I think that's exactly what this PR proposes to do with `Public`. If we did create that separate role, I'm not sure what the point of `Public` would be.

Reply:
I was suggesting that  instead of allowing Public role access to anything in  Airflow that we change the signup role to be Viewer, say.

```
AUTH_USER_REGISTRATION_ROLE = 'Public'
```

Reply:
@ashb , I don't think we want the viewer as the default role as the viewer could view all the dags info unless we change this assumption. I think having public role as default while enabling the role to access the default home page seems to be better.  The use case we have is that we only want certain dags with the source code to be viewer by a certain group user as those are high sensitive dags. 

Later if users want to access certain dags, they could create the respective dag role for those users.

Reply:
Right, I guess that makes sense too.

I just feel a little bit uneasy about making the home page/`@has_access` accessible to Public role (i.e. anon users), (even if not l no dags are shown) in a default install, and would rather it was at least a config option of some sort.

Reply:
@ashb, could you elaborate your concern? Currently this should just let the user know that they need to register with certain type of roles to view the dag instead of blank page. I think this will help to make the onboarding for a new user smoother.

Reply:
The problem with using "Public" role is that that applies to signed in users _and_ not-signed in users. That feels I think is my cause for concern, as it's conflating two concepts.

My concern is around "defaults" for an Airflow install - they should be, by default, locked down tight.

Most people won't have  registration enabled, nor will they want it, so display the home page, even with no DAGs to an anon user just feels wrong and a confusing/worrying, if not _strictly_ a security problem.

I think if we didn't use the Public role here the changes to the home-page etc are not a problem, though I'd like to get rid of `is_user_logged_in` being passed to the home page template, as I don't want that being shown to an not-logged-in user - I prefer the existing behaviour of redirecting to a login page.